Modules/binascii.c

```diff
@@ -510,7 +510,16 @@ binascii_a2b_base64_impl(PyObject *module, Py_buffer *data)
     }
 
     if (leftbits != 0) {
-        PyErr_SetString(Error, "Incorrect padding");
+        if (leftbits == 6) {
+            /*
+            ** There is exactly one extra valid, non-padding, base64 character.
+            ** This is an invalid length, as there is no possible input that
+            ** could encoded into such a base64 string.
+            */
+            PyErr_SetString(Error, "Invalid length");
```

Comment:
If we're going to change this, I'd suggest adding some verbiage to the error message saying that padding ==6 (or whatever) is invalid in base64, and probably mentioning the value of leftbits, if that's useful to the caller. But just having two different error messages for different cases with no additional context would confuse me more than it would help me.

Reply:
Something like this? "Invalid base64-encoded string length: remainder when dividing length by 4 must be 0, 2 or 3"

Reply:
Note that it is important, IMO, that the messages be different.  In the 'incorrect padding' case one can add padding to get a decode (which may be wrong, but most likely will be right).  In the invalid length case, one cannot do that.  The email package cares about this, as it tries its best to do the decode even if the padding is incorrect.  I'm pretty sure there is an open issue against the email package for which this is the answer, though I don't have time to go look  for it right now :)

Reply:
@bitdancer, were you referring to [bpo-27397](https://bugs.python.org/issue27397)?

Reply:
Also, if this difference is important, perhaps this should be a separate exception class/sub-class?